src/InvestJob.jsx

```diff
@@ -276,103 +264,21 @@ export class InvestJob extends React.Component {
-    // Right now we're never calling validate with a limit_to,
-    // but we have an awful lot of logic here to cover it.
-    if (limit_to) {
-      payload['limit_to'] = limit_to
-    }
-
-    const results = await fetchValidation(payload);
-
-    // A) At least one arg was invalid:
-    if (results.length) { 
-
-      results.forEach(result => {
-        // Each result is an array of two elements
-        // 0: array of arg keys
-        // 1: string message that pertains to those args
-        const argkeys = result[0];
-        const message = result[1];
-        argkeys.forEach(key => {
-          argsMeta[key]['validationMessage'] = message
-          argsMeta[key]['valid'] = false
-          keyset.delete(key);
-        })
-      });
-      if (!limit_to) {  // validated all, so ones left in keyset are valid
-        keyset.forEach(k => {
-          argsMeta[k]['valid'] = true
-          argsMeta[k]['validationMessage'] = ''
-        })
-      }
-      this.setState({
-        args: argsMeta,
-        argsValid: false
-      });
-
-    // B) All args were validated and none were invalid:
-    } else if (!limit_to) {
-      
-      keyset.forEach(k => {
-        argsMeta[k]['valid'] = true
-        argsMeta[k]['validationMessage'] = ''
-      })
-      // It's possible all args were already valid, in which case
-      // it's nice to avoid the re-render that this setState call
-      // triggers. Although only the Viz app components re-render 
-      // in a noticeable way. Due to use of redux there?
-      if (!this.state.argsValid) {
-        this.setState({
-          args: argsMeta,
-          argsValid: true
-        })
-      }
-
-    // C) Limited args were validated and none were invalid
-    } else if (limit_to) {
-
-      argsMeta[limit_to]['valid'] = true
-      // this could be the last arg that needed to go valid,
-      // in which case we should trigger a full args_dict validation
-      // in order to properly set state.argsValid
-      this.setState({ args: argsMeta },
-        () => {
-          let argIsValidArray = [];
-          for (const key in argsMeta) {
-            argIsValidArray.push(argsMeta[key]['valid'])
-          }
-          if (argIsValidArray.every(Boolean)) {
-            this.investValidate(argsValuesFromSpec(argsMeta));
-          }
-        }
-      );
-    }
-  }
-
-  async investGetSpec(modelName) {
+  async investGetSpec(modelName, argsInitDict={}) {
     /** Get an invest model's ARGS_SPEC when a model button is clicked.
-    * A side-effect is that much of this component's state is reset.
+    *  
+    * Also get the model's UI spec if it exists.
+    * Then reset much of this component's state in case a prior job's 
+    * state exists. This includes setting a new setupKey, which is passed
+    * as a key to the SetupTab component, triggering it to re-mount
+    * rather than just re-render, allowing one-time initilization of
+    * arg grouping and ordering.

```

Comment:
Ah interesting. I wonder if it'd be useful to have an actual ID of some sort for every time a model button is clicked... maybe not.

Reply:
I think the main requirement is just that the key changes. At first I was re-generating a random string, then I thought I would hash the ARGS_SPEC + any initial arg values, then I ended up just incrementing an integer.